rotkehlchen/db/optimismtx.py

```diff
@@ -0,0 +1,107 @@
+import logging
+from typing import TYPE_CHECKING, Any, Optional
+
+from rotkehlchen.chain.optimism.types import OptimismTransaction
+from rotkehlchen.db.evmtx import DBEvmTx
+from rotkehlchen.db.filtering import OptimismTransactionsFilterQuery
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_timestamp
+from rotkehlchen.types import ChainID, ChecksumEvmAddress, deserialize_evm_tx_hash
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.drivers.gevent import DBCursor
+
+from rotkehlchen.constants.limits import FREE_ETH_TX_LIMIT
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class DBOptimismTx(DBEvmTx):
+
+    def add_optimism_transactions(
+            self,
+            write_cursor: 'DBCursor',
+            optimism_transactions: list[OptimismTransaction],
+            relevant_address: Optional[ChecksumEvmAddress],
+    ) -> None:
+        """Adds optimism transactions to the database"""
+        self.add_evm_transactions(
+            write_cursor,
+            optimism_transactions,  # type: ignore[arg-type]
+            relevant_address,
+        )
+
+        tx_tuples: list[tuple[Any, ...]] = []
+        for tx in optimism_transactions:
+            tx_tuples.append((
+                tx.tx_hash,
+                tx.chain_id.serialize_for_db(),
+                tx.l1_fee,
+            ))
+
+        query = """
+            INSERT INTO optimism_transactions(
+              tx_hash,
+              chain_id,
+              l1_fee)
+            VALUES (?, ?, ?)
+        """
+        self.db.write_tuples(
+            write_cursor=write_cursor,
+            tuple_type='evm_transaction',
+            query=query,
+            tuples=tx_tuples,
+            relevant_address=relevant_address,
+        )
+
+    def get_optimism_transactions(
+            self,
+            cursor: 'DBCursor',
+            filter_: OptimismTransactionsFilterQuery,
+            has_premium: bool,
+    ) -> list[OptimismTransaction]:
+        """Returns a list of optimism transactions optionally filtered by
+        the given filter query
+
+        This function can raise:
+        - pysqlcipher3.dbapi2.OperationalError if the SQL query fails due to invalid
+        filtering arguments.
+        """
+        query, bindings = filter_.prepare()
+        if has_premium:
+            query = 'SELECT DISTINCT evm_transactions.tx_hash, evm_transactions.chain_id, timestamp, block_number, from_address, to_address, value, gas, gas_price, gas_used, input_data, nonce, l1_fee FROM (SELECT * FROM evm_transactions WHERE chain_id = 10) AS evm_transactions LEFT JOIN optimism_transactions on evm_transactions.tx_hash = optimism_transactions.tx_hash ' + query  # noqa: E501
```

Comment:
you are duplicating queries here. This is not how you join. You are making an unneeded subquery. What you should write is something more like.

```
SELECT DISTINCT EVM.tx_hash, 10, EVM.timestamp, EVM.block_number, EVM.from_address, EVM.to_address, EMV.value, EVM.gas, EVM.gas_price, EVM.gas_used, EVM.input_data, EVM.nonce, OP.l1_fee FROM evm_transactions AS EVM LEFT JOIN optimism_transactions AS OP ON EVM.tx_hash=OP.tx_hash
```